raster-foundry/raster-foundry
# Use annotations API on the frontend

## Overview
Update annotations API with changes required to support frontend functions efficiently
Update frontend to use the annotations api using redux
Convert some files to single file components

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

## Testing Instructions
* On the annotations page:
  - Create a new annotation, using each of the shapes
  - Verify that editing works for all of them (shape and metadata)
* Delete an annotation
* Test all the hotkeys
* Verify that import / exporting still works
* Verify that the clear annotations button works, and has a warning
* Verify that filtering works
* Verify that suggestions when selecting a label work
* Verify that if you don't label an annotation, it gets filed as "Unlabeled" and a filter for "Unlabeled" annotations appears
* Verify that all changes that are made get saved to the api as soon as you click save

Closes #2740 
Closes #2710
